app/main.py

```diff
@@ -1,3 +1,15 @@
-def cache(func):
-    # Write your code here
-    pass
+from typing import Callable, Any
+
+
+def cache(func: Callable) -> Any:
+    data = {}
+
+    def inner(*args) -> Any:
```

Comment:
add typehint for args

Reply:
"functions that take immutable arguments"
int, float, bool, tuple....
But how can I put it all together?

Reply:
> add typehint for args

"functions that take immutable arguments"
int, float, bool, tuple....
But how can I put it all together?

Reply:
"functions that take immutable arguments"
int, float, bool, tuple....
But how can I put it all together?

Reply:
from typing import **Any**